src/util.ts

```diff
@@ -543,4 +534,26 @@ export function checkFileExists(filePath: string): Promise<boolean> {
           }
       });
   });
+}
+
+export function disposeAll(disp: Iterable<vscode.Disposable>) {
+  for (const d of disp) {
+    d.dispose();
+  }
+}
+
+export function reportProgress(progress: ProgressHandle|undefined, message: string) {
+  if (progress) {
+    progress.report({message});
+  }
+}
+
+export function chokidarOnAnyChange(watcher: chokidar.FSWatcher, listener: (path: string, stats?: fs.Stats | undefined) => void) {
+  return watcher.on('add', listener)
+                .on('change', listener)
+                .on('unlink', listener);
+}
+
+export function isString(x: any) {
```

Comment:
This is how we do it in cpptools:
```
export function isString(input: any): input is string {
    return typeof(input) === "string";
}
```
This has the benefit of also promoting `input` to `string` for anything inside the block after `if (isString(x))`

Reply:
This version in cpptools doesn't work for string objects, i.e., typeof(new String('whatever')) is object instead of string.

Reply:
But String objects should never be used and are condemned as bad practice.  I didn't see `new String` anywhere in the code. Is VS Code sending us String objects?  If we want to be safe against String objects, the code should be:

```ts
export function isString(input: any): input is string {
    return typeof(input) === "string" || input instanceof String;
}
```

Reply:
Not sure if this argument worth it but I was following the Array.IsArray implementation in the standard... The second version you provided doesn't work for String.prototype...

Reply:
We should never have String objects and certainly should not be manipulating the prototype, so I don't think we need this level of protection, but if you feel strongly about it, you can leave it.  Just add the `x is string` part after the signature so that the type will be promoted to string by IntelliSense.